compiler/rustc_typeck/src/collect.rs

```diff
@@ -227,11 +228,30 @@ fn reject_placeholder_type_signatures_in_item(tcx: TyCtxt<'tcx>, item: &'tcx hir
         // `static`, `fn` and `const` are handled elsewhere to suggest appropriate type.
         _ => return,
     };
+    let kind = match &item.kind {
+        hir::ItemKind::Union(..) => "union",
+        hir::ItemKind::Enum(..) => "enum",
+        hir::ItemKind::TraitAlias(..) => "trait alias",
+        hir::ItemKind::Trait(..) => "trait",
+        hir::ItemKind::Impl(..) => "implementation",
+        hir::ItemKind::Struct(..) => "struct",
+        hir::ItemKind::OpaqueTy(..) => "opaque type",
+        hir::ItemKind::TyAlias(..) => "type alias",
+        _ => return,
+    };
```

Comment:
Could you move this to `impl hir::ItemKind` and make the wording compatible with `ast::ItemKind::descr`?